### ISSUE TITLE
`ChooseVersion` doesn't match recent Keystone builds

### ISSUE THREAD

**Issue by Quentin-M** [gophercloud/gophercloud#106]
Hi,

[ChooseVersion](https://github.com/gophercloud/gophercloud/blob/0bc5578dc193f747667cdeee035460f4a168e015/openstack/utils/choose_version.go#L23), as it is used in [Authenticate](https://github.com/gophercloud/gophercloud/blob/0bc5578dc193f747667cdeee035460f4a168e015/openstack/client.go#L65) tries to explicitly match the `ID` field of the Keystone response to `v2.0` and `v3.0`.

However, my Keystone API returns the following:

```
{
    "versions": {
        "values": [
            {
                "status": "stable",
                "updated": "2016-10-06T00:00:00Z",
                "media-types": [
                    {
                        "base": "application/json",
                        "type": "application/vnd.openstack.identity-v3+json"
                    }
                ],
                "id": "v3.7",
                "links": [
                    {
                        "href": "http://identity.openstack.cluster:30080/v3/",
                        "rel": "self"
                    }
                ]
            },
            {
                "status": "deprecated",
                "updated": "2016-08-04T00:00:00Z",
                "media-types": [
                    {
                        "base": "application/json",
                        "type": "application/vnd.openstack.identity-v2.0+json"
                    }
                ],
                "id": "v2.0",
                "links": [
                    {
                        "href": "http://identity.openstack.cluster:30080/v2.0/",
                        "rel": "self"
                    },
                    {
                        "href": "http://docs.openstack.org/",
                        "type": "text/html",
                        "rel": "describedby"
                    }
                ]
            }
        ]
    }
}
```

The v3 API can't be matched due to the ID being `v3.7` and the v2 API can't either as it is deprecated.

**Comment by jrperritt** [comment]
`Authenticate` is more of a convenience function (though it probably doesn't seem convenient to you at the moment). It may now make sense to parse the version `id`s and look for something reasonable. I suspect I'll probably not get around to doing that (but I'd be willing to review a PR for it). However, if you know you want to use v3, the suggested way is to create a new client and directly authenticate to v3:
```go
pc, err := gophercloud.NewClient("authEndpoint")
err := gophercloud.AuthenticateV3(pc, ao, eo)
```

**Comment by jtopjian** [comment]
This was resolved in #508. However, if you're still running into problems, let me know.
